### PR TITLE
Address Wdeprecated-copy warnings on gcc 9.3

### PR DESCRIPTION
**Pull request by DVigneault**
This addresses warnings in itk::MapContainer and
itk::VectorContainer regarding deprecation of implicitly defined
operator= (with explicitly defined copy constructor) or vice
versa.  This begins to address GitHub Issue #1302.  There
remain a number of warnings in SpatialObjects, so this patch
does not completely resolve the issue.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [X] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)
- [x] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by DVigneault** [comment]
There are a couple very minor bug fixes buried in the second commit:

1. `itk::DTITubeSpatialObjectPoint` specified the wrong `Superclass`.
2. `itk::LineSpatialObjectPoint` and `itk::SurfaceSpatialObjectPoint` did not check if `this != &rhs` in `operator=`

**Comment by DVigneault** [comment]
@phcerdan Thanks for the comments!  Will work on this and push another commit later tonight.

File: Modules/Core/SpatialObjects/include/itkDTITubeSpatialObjectPoint.h [hunk 1]
```diff
@@ -61,7 +61,7 @@ class ITK_TEMPLATE_EXPORT DTITubeSpatialObjectPoint : public TubeSpatialObjectPo
 {
 public:
   using Self = DTITubeSpatialObjectPoint;
-  using Superclass = SpatialObjectPoint<TPointDimension>;
+  using Superclass = TubeSpatialObjectPoint<TPointDimension>;
```
Comment (phcerdan) [inline Modules/Core/SpatialObjects/include/itkDTITubeSpatialObjectPoint.h:64]: Good spotting.

File: Modules/Core/Common/include/itkMapContainer.h [hunk 2]
```diff
@@ -275,6 +275,9 @@ class ITK_TEMPLATE_EXPORT MapContainer
     ConstIterator(const Iterator & r)
       : m_Iter(r.m_Iter)
     {}
+    ConstIterator(const ConstIterator & r)
```
Comment (N-Dekker) [inline Modules/Core/Common/include/itkMapContainer.h:278]: Thank you Davis, But could this copy-constructor simply be defaulted? Could both copy-constructor and assignment operator be defaulted _implicitly_, according to the Rule Of Zero? If so, that would be my preference.

Comment (DVigneault) [reply Modules/Core/Common/include/itkMapContainer.h]: @N-Dekker Thank you for the review!  You're absolutely right--please see new commits.

**Comment by DVigneault** [comment]
- Added tests for `itk::VectorContainer`
- Reordered commits slightly into a more logical progression
- Removed `WIP`

**Pull request merged**